File: OrderControlSystem.UI/src/stores/account.js
import { defineStore } from "pinia";
import api from "@/api";
import qs from "qs";

export const useAccountStore = defineStore({
  id: "account",

  state: () => ({
    accounts: [],
    accountLoading: false,
  }),

  getters: {
    getAccounts: (state) => state.accounts,
  },

  actions: {
    async fetchAccounts() {
      this.accountLoading = true;
      let res = await api.get("Account/GetAllAccount");
     this.accounts=res.data?.value;
      this.accountLoading = false;
    },

    async updateAccount(accountDto) {
      this.accountLoading = true;
      await api.post("Account/UpdateAccount", accountDto);
      this.accountLoading = false;
    },

    async addAccount(account) {
      this.accountLoading = true;
      let accountQs = {
        customerId: account.customerId,
        username: account.username,
        password: account.password,
        role: account.role,
        name: account.name,
        lastName: account.lastName,
        email: account.email,
        phoneNumber: account.phoneNumber,
        isActive: account.isActive,
      };
      await api.post("Account/CreateAccount", accountQs);
      this.accountLoading = false;
    },

    async deleteAccount(account) {
      this.accountLoading = true;
      let accountQs = qs.stringify({
        accountId: account.accountId,
      });
      await api.post("Account/DeleteAccount", accountQs);
      this.accountLoading = false;
    },
  },
  persist: false,
});

export default useAccountStore;

File: OrderControlSystem.DAL/Account.cs
using System;

#nullable disable

namespace OrderControlSystem.DAL
{
    public partial class Account
    {
        public int Id { get; set; }
        public int? CustomerId { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public short Role { get; set; }
        public string Name { get; set; }
        public string LastName { get; set; }
        public DateTime CreateDati { get; set; }
        public DateTime? UpdateDati { get; set; }
        public bool? IsActive { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }

        public virtual Customer Customer { get; set; }
        public virtual Role RoleNavigation { get; set; }
    }
}
